server/src/test/java/org/apache/cassandra/sidecar/routes/NodeDecommissionHandlerTest.java

```diff
@@ -0,0 +1,216 @@
+        .when(mockStorageOperations).decommission(anyBoolean());
+
+        WebClient client = WebClient.create(vertx);
+        String testRoute = "/api/v1/cassandra/operations/decommission";
+        client.put(server.actualPort(), "127.0.0.1", testRoute)
+              .expect(ResponsePredicate.SC_ACCEPTED)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(ACCEPTED.code());
+                  OperationalJobResponse decommissionResponse = response.bodyAsJson(OperationalJobResponse.class);
+                  assertThat(decommissionResponse).isNotNull();
+                  assertThat(decommissionResponse.status()).isEqualTo(RUNNING);
+                  context.completeNow();
+              }));
+    }
+
+    @Test
+    void testDecommissionCompleted(VertxTestContext context)
+    {
+        when(mockStorageOperations.getOperationMode()).thenReturn("NORMAL");
+        WebClient client = WebClient.create(vertx);
+        String testRoute = "/api/v1/cassandra/operations/decommission";
+        client.put(server.actualPort(), "127.0.0.1", testRoute)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(OK.code());
+                  LOGGER.info("Decommission Response: {}", response.bodyAsString());
+
+                  OperationalJobResponse decommissionResponse = response.bodyAsJson(OperationalJobResponse.class);
+                  assertThat(decommissionResponse).isNotNull();
+                  assertThat(decommissionResponse.status()).isEqualTo(SUCCEEDED);
+                  context.completeNow();
+              }));
+    }
+
+    @Test
+    void testDecommissionFailed(VertxTestContext context)
+    {
+        when(mockStorageOperations.getOperationMode()).thenReturn("NORMAL");
+        doThrow(new RuntimeException("Simulated failure")).when(mockStorageOperations).decommission(anyBoolean());
+        WebClient client = WebClient.create(vertx);
+        String testRoute = "/api/v1/cassandra/operations/decommission";
+        client.put(server.actualPort(), "127.0.0.1", testRoute)
+              .expect(ResponsePredicate.SC_OK)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(OK.code());
+                  context.completeNow();
+              }));
+    }
+
+    @Test
+    void testDecommissionConflict(VertxTestContext context)
+    {
+        when(mockStorageOperations.getOperationMode()).thenReturn("LEAVING");
+        WebClient client = WebClient.create(vertx);
+        String testRoute = "/api/v1/cassandra/operations/decommission";
+        client.put(server.actualPort(), "127.0.0.1", testRoute)
+              .expect(ResponsePredicate.SC_CONFLICT)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(CONFLICT.code());
```

Comment:
nit: add the assertion for the returned jobId, so that client can use this jobId to track the existing inflight job.